### PR TITLE
indexer: fix upgrade when password contains a %

### PR DESCRIPTION
**Pull request by sileht**
oslo.config doesn't support %{} or %() variable interpolation while
ConfigParser does.

So when a password contains a '%', it's fine for oslo.config
But the Python ConfigParser will raise the exception.

Since alembic use ConfigParser and not oslo.config, we have to escape
the url ourself.

This is not a big deal since, we don't want alembic doing variable
interpolation.

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 9b3cf5dd22305fd52b9d362b50eea72d645f26bb

File: gnocchi/tests/indexer/sqlalchemy/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2015 eNovance
```
Comment (jd) [inline gnocchi/tests/indexer/sqlalchemy/test_utils.py]: good one

Comment (sileht) [reply gnocchi/tests/indexer/sqlalchemy/test_utils.py]: done

**Comment by sileht** [comment]
Pull-request updated, HEAD is now cb610f7c3934b39afa834654b0ef22519a37b93c

**Pull request merged**